src/sorted_set.jl

```diff
@@ -180,8 +180,7 @@ end
 Returns the key type for SortedSet.
 This function may also be applied to the type itself. Time: O(1)
 """
-@inline eltype(m::SortedSet{K,Ord}) where {K,Ord <: Ordering} = K
-@inline eltype(::Type{SortedSet{K,Ord}}) where {K,Ord <: Ordering} = K
+@inline Base.eltype(::Type{SortedSet{K,Ord}}) where {K,Ord <: Ordering} = K
```

Comment:
Why are you proposing to remove the method of  `eltype`  that accepts a sorted set instead of its type?  Here is what the documentation (1.1) says about `eltype`:

***Determine the type of the elements generated by iterating a collection of the given type. For dictionary types, this will be a Pair{KeyType,ValType}. The definition eltype(x) = eltype(typeof(x)) is provided for convenience so that instances can be passed instead of types. However the form that accepts a type argument should be defined for new types.***


So according to the docs, it is OK to define a version of `eltype` that takes the container itself.  By removing this statement, you may break people's code that uses this functionality.  Furthermore, `SortedSet` is intended to a mostly drop-in replacement for `Set`, and `Set` supports both `eltype` calls.

Reply:
`eltype` of an instance falls back to `eltype(typeof(x))` so this won't break any code.

Reply:
Got it, thanks!

Reply:
I actually included tests for this as well cause I was worried about the same thing.